### ISSUE TITLE
GCI diffs are not correctly reported by the JSON printer

### ISSUE THREAD

**Issue by wyuenho** [golangci/golangci-lint#6296]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

I don't know

### Description of the problem

When invoking `gci main.go` directly, the following is emitted:

```console
--- main.go
+++ main.go
@@ -1,8 +1,8 @@
 package main
 
 import (
-	"os"
 	"fmt"
+	"os"
 )
 
 func main() {

```

However, when calling `golangci-ling run --output.json.path stdout`, only the deletion is emitted but not the addition.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.7.2 built with go1.25.4 from 9f61b0f5 on 2025-12-07T16:57:12Z
```

</details>

### Configuration

<details>

```yaml
version: '2'
formatters:
  enable:
    - gci
  settings:
    gci:
      sections:
        - standard
        - default
      custom-order: true

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.25.5 linux/amd64
AR='ar'
CC='gcc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='g++'
GCCGO='gccgo'
GO111MODULE=''
GOAMD64='v1'
GOARCH='amd64'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/home/jwong/.cache/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/home/jwong/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS='-buildvcs=false'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build1941364073=/tmp/go-build -gno-record-gcc-switches'
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMOD='/home/jwong/golangci-lint-bug/go.mod'
GOMODCACHE='/home/jwong/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/jwong/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/snap/go/11017'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/home/jwong/.config/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/snap/go/11017/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.25.5'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v --output.json.path stdout --show-stats=false --max-issues-per-linter=0 --max-same-issues=0 --path-mode=abs main.go  | jq '.Issues[]?.SuggestedFixes[]?.TextEdits[]?.NewText |= if . != null then @base64d else . end | .Issues[]'
level=info msg="golangci-lint has version 2.7.2 built with go1.25.4 from 9f61b0f5 on 2025-12-07T16:57:12Z"
level=info msg="[config_reader] Config search paths: [./ /home/jwong/core3/src/experimental/jwong /home/jwong/golangci-lint /home/jwong /home /]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[config_reader] Module name \"golangci-lint\""
level=info msg="maxprocs: Leaving GOMAXPROCS=20: CPU quota undefined"
level=info msg="[goenv] Read go env for 62.544995ms: map[string]string{\"GOCACHE\":\"/home/jwong/.cache/go-build\", \"GOROOT\":\"/snap/go/11017\"}"
level=info msg="[lintersdb] Active 6 linters: [errcheck gci govet ineffassign staticcheck unused]"
level=info msg="[loader] Go packages loading at mode 8767 (deps|files|imports|name|types_sizes|compiled_files|exports_file) took 173.339819ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 84.555µs"
level=info msg="[linters_context/goanalysis] analyzers took 758.219597ms with top 10 stages: buildir: 593.286444ms, inspect: 31.339519ms, ctrlflow: 29.147257ms, printf: 20.001074ms, fact_deprecated: 19.329997ms, SA5012: 18.413843ms, fact_purity: 16.016298ms, nilness: 15.582764ms, typedness: 9.846857ms, tokenfileanalyzer: 2.946328ms"
level=info msg="[runner] Issues before processing: 2, after processing: 1"
level=info msg="[runner] Processors filtering stat (in/out): source_code: 1/1, severity-rules: 1/1, sort_results: 1/1, path_absoluter: 2/2, path_relativity: 2/2, nolint_filter: 2/2, max_per_file_from_linter: 2/1, max_same_issues: 1/1, path_prettifier: 1/1, filename_unadjuster: 2/2, generated_file_filter: 2/2, diff: 2/2, fixer: 2/2, path_shortener: 1/1, invalid_issue: 2/2, exclusion_rules: 2/2, max_from_linter: 1/1, cgo: 2/2, exclusion_paths: 2/2, uniq_by_line: 2/2"
level=info msg="[runner] processing took 60.568µs with stages: nolint_filter: 21.816µs, generated_file_filter: 18.515µs, source_code: 8.962µs, path_relativity: 2.684µs, uniq_by_line: 1.551µs, sort_results: 1.03µs, path_shortener: 874ns, max_per_file_from_linter: 841ns, cgo: 661ns, invalid_issue: 541ns, max_same_issues: 506ns, path_prettifier: 410ns, exclusion_rules: 384ns, exclusion_paths: 384ns, fixer: 364ns, filename_unadjuster: 309ns, path_absoluter: 264ns, diff: 253ns, max_from_linter: 131ns, severity-rules: 88ns"
level=info msg="[runner] linters took 547.788922ms with stages: goanalysis_metalinter: 547.669715ms"
level=info msg="File cache stats: 1 entries of total size 80B"
{
  "FromLinter": "gci",
  "Text": "File is not properly formatted",
  "Severity": "",
  "SourceLines": [
    "\t\"os\""
  ],
  "Pos": {
    "Filename": "main.go",
level=info msg="Memory: 10 samples, avg is 147.5MB, max is 295.2MB"
    "Offset": 23,
    "Line": 4,
    "Column": 1
level=info msg="Execution took 847.658663ms"
  },
  "SuggestedFixes": [
    {
      "Message": "",
      "TextEdits": [
        {
          "Pos": 23,
          "End": 28,
          "NewText": ""
        }
      ]
    }
  ],
  "ExpectNoLint": false,
  "ExpectedNoLintLinter": ""
}

```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// main.go

package main

import (
	"os"
	"fmt"
)

func main() {
	fmt.Println(os.Getwd())
}

```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
This is because the number of reports from a formatter are limited when using `run` command.
This is expected and hardcoded.

**Comment by wyuenho** [comment]
From `golangci-lint run --help`

```
...
      --max-issues-per-linter int         Maximum issues count per one linter. Set to 0 to disable (default 50)
      --max-same-issues int               Maximum count of issues with the same text. Set to 0 to disable (default 3)
```

I've tried setting both to 0 and leaving them as the default. In both cases I got 1 report only, but --fix is able to move the line. This is not expected.

**Comment by ldez** [comment]
The limitation is **hardcoded** for formatters, you can not change it.
This limitation is related to how formatters differ from linters.
The behavior is expected because of that.

**Comment by wyuenho** [comment]
Sorry I'm not sure I follow, do you care to elaborate why there is such a hardcoded limit? Is it because golangci-lint doesn't have the capability to linearize issues reported by multiple formatters and resolve conflicts, and instead of reporting an error, you've chosen to silently truncate the report instead?
